src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -965,32 +962,27 @@ private References LoadAssemblies(IEnumerable<string> refs, bool loadTestNames,
         /// Does *not* close the given memory stream, and
         /// returns true if the serialization has been successfully generated.
         /// </summary>
-        private bool SerializeSyntaxTree(MemoryStream ms)
+        private bool WriteSyntaxTreeSerialization(MemoryStream ms)
         {
             void LogError() => this.LogAndUpdate(
                 ref this.compilationStatus.Serialization, ErrorCode.SerializationFailed, Enumerable.Empty<string>());
 
-            this.compilationStatus.Serialization = 0;
-            if (this.CompilationOutput == null)
+            void LogExceptionAndError(Exception ex)

```

Comment:
This looks like it is still work in progress, so I'll take a look when it's done.

Reply:
Ready to review!